spec/rails_helper.rb

```diff
@@ -518,6 +518,22 @@ class SpecTimeoutError < StandardError
         sleep 0.01 while !mutex.synchronize { is_waiting }
       end
     end
+
+    config.around do |example|
+      example.run
+
+      if example.exception.is_a?(Socket::ResolutionError)
+        require "socket"
```

Comment:
Not a blocker, but curious: shouldn't `Socket` be available if we're referencing `Socket::ResolutionError` in the conditional?

Reply:
Good point :+1